cmd/skaffold/app/cmd/cmd.go

```diff
@@ -44,47 +46,68 @@ var (
 
 func NewSkaffoldCommand(out, err io.Writer) *cobra.Command {
 	updateMsg := make(chan string)
+	var shutdownAPIServer func() error
 
 	rootCmd := &cobra.Command{
 		Use:           "skaffold",
 		Short:         "A tool that facilitates continuous development for Kubernetes applications.",
 		SilenceErrors: true,
-	}
-
-	rootCmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
-		opts.Command = cmd.Use
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			cmd.Root().SilenceUsage = true
 
-		if err := SetUpLogs(err, v); err != nil {
-			return err
-		}
+			opts.Command = cmd.Use
 
-		if forceColors {
-			color.ForceColors()
-		}
+			// Setup colors
+			if forceColors {
+				color.ForceColors()
+			}
+			color.OverwriteDefault(color.Color(defaultColor))
 
-		rootCmd.SilenceUsage = true
-		logrus.Infof("Skaffold %+v", version.Get())
-		color.OverwriteDefault(color.Color(defaultColor))
+			// Setup logs
+			if err := setUpLogs(err, v); err != nil {
+				return err
+			}
 
-		if quietFlag {
-			logrus.Debugf("Update check is disabled because of quiet mode")
-		} else {
-			go func() {
-				if err := updateCheck(updateMsg); err != nil {
-					logrus.Infof("update check failed: %s", err)
-				}
-			}()
-		}
+			// Start API Server
+			if cmd.Use == "dev" {
+				// TODO(dgageot): api server is always started in dev mode, right now.
+				// It should instead default to true.
```

Comment:
do you mean default to false, if the api server is always started right now?

Reply:
I mean it should default to true for dev mode but the user should be able to disable it. Currently, it's always on for dev mode

Reply:
I see, thanks!

Reply:
how about either
* changing the `--enable-rpc` flag to a `*bool`, so we can tell if the user didn't provide the flag (vs. explicitly providing `--enable-rpc=false`, or
* adding a separate flag `--disable-rpc`, and checking that here e.g. `if cmd.Use == "dev" && opts.DisableRPC == false`. we'd have to have a way to make sure users don't set `--enable-rpc` and `--disable-rpc` at the same time though (don't know if there's an easy way to do this using cobra)

I'm fine with this being out of scope for this PR though.